src/json.c

```diff
@@ -545,6 +545,7 @@ lyjson_object(struct lyjson_ctx *jsonctx)
     LY_CHECK_RET(skip_ws(jsonctx));
 
     if (*jsonctx->in->current == '}') {
+        jsonctx->depth = jsonctx->depth ? jsonctx->depth - 1 : 0;
```

Comment:
Is the check for `0` really needed? How can the depth be 0 if the closing curly bracket was found? Or is this before the validation of these brackets is performed? But it seems it should be before.

Reply:
Just put an `assert()` there if it should never be `0` to prevent possible underflow.

Reply:
The point is that we are talking strictly about JSON here, which must have `}` for each `{`, otherwise it is not valid JSON. And invalid JSON should be detected in the generic JSON parser (`json.c`) and not left up to the higher-level parsers. But if you check where is `lyjson_object()` called, you will see that only after a `{` was parsed so I believe `assert()` is the way to go.